frontend/routes.js

```diff
@@ -9,17 +9,26 @@ import SiteBuildLogs from './components/site/siteBuildLogs';
 import SiteUsers from './components/site/SiteUsers';
 import SitePublishedBranchesTable from './components/site/sitePublishedBranchesTable';
 import SitePublishedFilesTable from './components/site/sitePublishedFilesTable';
+import SiteGitHubBranches from './components/site/SiteGitHubBranches';
 import SiteSettings from './components/site/SiteSettings';
 import AddSite from './components/AddSite';
 import NotFound from './components/NotFound';
 
+import siteActions from './actions/siteActions';
+import userActions from './actions/userActions';
+
+const fetchInitialData = () => {
+  userActions.fetchUser();
+  siteActions.fetchSites();
+};
+
 export default (
-  <Route path="/" component={App}>
+  <Route path="/" component={App} onEnter={fetchInitialData}>
```

Comment:
I found out about this `onEnter` callback while reading the react-router docs and thought this was a cleaner place to initiate the data fetch than where it was in `main.jsx`

Reply:
Nice, that's fancy 🤵